File: coding_round2_struct/Safe_Night_Cab/starter_code/solution.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        if (!sc.hasNextInt()) return;

        int n = sc.nextInt();
        long d = sc.nextLong();
        long r = sc.nextLong();

        long[] p = new long[n];
        for (int i = 0; i < n; i++) p[i] = sc.nextLong();

        // TODO: determine if travel from 0 to d is possible using illuminated segments
        // Print "YES" or "NO".
        // System.out.println("YES");
    }
}

File: coding_round2_struct/Safe_Night_Cab/starter_code/solution.c
#include <stdio.h>
#include <stdlib.h>

int main(void) {
    int n;
    long long d, r;
    if (scanf("%d %lld %lld", &n, &d, &r) != 3) return 0;

    long long *p = (long long *)malloc((size_t)n * sizeof(long long));
    if (!p) return 1;
    for (int i = 0; i < n; i++) scanf("%lld", &p[i]);

    // TODO: determine if travel from 0 to d is possible using illuminated segments
    // Print "YES" or "NO".
    // printf("YES\n");

    free(p);
    return 0;
}
